File: library/src/main/java/com/gmail/brymher/materialcalendar/format/DayFormatter.java
package com.gmail.brymher.materialcalendar.format;


import androidx.annotation.NonNull;

import com.gmail.brymher.materialcalendar.CalendarDay;
import com.gmail.brymher.materialcalendar.MaterialCalendarView;

import java.text.SimpleDateFormat;

/**
 * Supply labels for a given day. Default implementation is to format using a {@linkplain SimpleDateFormat}
 */
public interface DayFormatter {

  /**
   * Default format for displaying the day.
   */
  String DEFAULT_FORMAT = "d";

  /**
   * Default implementation used by {@linkplain MaterialCalendarView}
   */
  DayFormatter DEFAULT = new DateFormatDayFormatter();

  /**
   * Format a given day into a string
   *
   * @param day the day
   * @return a label for the day
   */
  @NonNull
  String format(@NonNull CalendarDay day);
}

File: library/src/main/java/com/gmail/brymher/materialcalendar/format/DayFormatter.kt
package com.gmail.brymher.materialcalendar.format;

import com.gmail.brymher.materialcalendar.CalendarDay
import com.gmail.brymher.materialcalendar.MaterialCalendarView
import java.text.SimpleDateFormat


/**
 * Supply labels for a given day. Default implementation is to format using a [SimpleDateFormat]
 */
abstract class DayFormatter {
    /**
     * Format a given day into a string
     *
     * @param day the day
     * @return a label for the day
     */
    abstract fun format(day: CalendarDay): String

    companion object {
        /**
         * Default format for displaying the day.
         */
        const val DEFAULT_FORMAT = "d"

        /**
         * Default implementation used by [MaterialCalendarView]
         */

        val DEFAULT: DayFormatter = DateFormatDayFormatter()
    }
}

